src/parser/parser.h

```diff
@@ -476,6 +476,8 @@ class CVC4_PUBLIC Parser {
 
   /** Create a new CVC4 bound variable expression of the given type. */
   Expr mkBoundVar(const std::string& name, const Type& type);
+  /** as above, but using internal naming scheme */
+  Expr mkBoundVar(const Type& type);
```

Comment:
Eliminate this function. It is not necessary to define the bound variable in the parser state since it is not being parsed.